src/MIDebugEngine/Engine.Impl/BreakpointManager.cs

```diff
@@ -24,6 +24,28 @@ public BreakpointManager(AD7Engine engine)
             _pendingBreakpoints = new System.Collections.Generic.List<AD7PendingBreakpoint>();
         }
 
+        private IEnumerable<AD7PendingBreakpoint> CodeBreakpoints
+        {
+            get
+            {
+                lock (_pendingBreakpoints)
```

Comment:
Since this code is doing deferred execution, the lock isn't going to do anything.